homeassistant/components/sensor/knx.py

```diff
@@ -0,0 +1,124 @@
+"""
+Sensors of a KNX Device.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/knx/
+"""
+from homeassistant.const import (TEMP_CELSIUS)
+from homeassistant.components.knx import (KNXConfig, KNXGroupAddress)
+
+
+DEPENDENCIES = ["knx"]
+
+# Speed units
+SPEED_METERPERSECOND = "m/s"  # type: str
+
+# Illuminance units
+ILLUMINANCE_LUX = "lx"  # type: str
+
+#  Predefined Minimum, Maximum Values for Sensors
+#  Temperature as defined in KNX Standard 3.10 - 9.001 DPT_Value_Temp
+KNX_TEMP_MIN = -273
+KNX_TEMP_MAX = 670760
+
+#  Luminance(LUX) as Defined in KNX Standard 3.10 - 9.004 DPT_Value_Lux
+KNX_LUX_MIN = 0
+KNX_LUX_MAX = 670760
+
+#  Speed m/s as defined in KNX Standard 3.10 - 9.005 DPT_Value_Wsp
+KNX_SPEED_MS_MIN = 0
+KNX_SPEED_MS_MAX = 670760
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Setup the KNX Sensor platform."""
+    # Add KNX Temperature Sensors
+    # KNX Datapoint 9.001 DPT_Value_Temp
+    if config["type"] == "temperature":
+        minimum_value, maximum_value = \
+            update_and_define_min_max(config, KNX_TEMP_MIN,
+                                      KNX_TEMP_MAX)
+
+        add_entities([
+            KNXSensorFloatClass(hass, KNXConfig(config), TEMP_CELSIUS,
+                                minimum_value, maximum_value)
+        ])
+
+    # Add KNX Speed Sensors(Like Wind Speed)
+    # KNX Datapoint 9.005 DPT_Value_Wsp
+    if config["type"] == "speed_ms":
```

Comment:
Don't put the units in there - as this will allow future implementation of configuring `units` for the speed

```python
CONF_SPEED = 'speed'
```

```python
elif config.get(CONF_TYPE) == CONF_SPEED:
```